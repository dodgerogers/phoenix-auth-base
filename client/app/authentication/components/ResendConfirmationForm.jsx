import React from 'react'
import { Field } from 'redux-form/immutable';
import { reduxForm } from 'redux-form/immutable';
import { Form, Message, Button, Icon } from 'semantic-ui-react';
import Input from '../../common/components/Input';
import { isRequired, isEmail } from '../../lib/utils/validation';
import { formIDs } from '../constants';


const ResendConfirmationForm = (props) => {
  const { handleSubmit, error, submitting, dirty } = props;

  return (
    <div className="resend-confirmation-form">
      <Form
        size='large'
        loading={submitting}
        onSubmit={handleSubmit}
      >
        {error && dirty && <Message negative>{error}</Message>}
        <Field
          component={Input}
          fluid
          name="email"
          icon="mail"
          iconPosition="left"
          placeholder="Email"
          validate={[isRequired, isEmail]}
        />
        <Button
          type="submit"
          fluid={true}
          color="teal"
          disabled={!props.valid}
        >
          Submit
        </Button>
      </Form>
    </div>
  );
}

export { ResendConfirmationForm as PureComponent };
export default reduxForm({ form: formIDs.RESEND_CONFIRMATION })(ResendConfirmationForm);
