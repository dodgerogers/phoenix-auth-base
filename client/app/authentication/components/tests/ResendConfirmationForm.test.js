import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ResendConfirmationForm from '../ResendConfirmationForm';


describe('ResendConfirmationForm', () => {
  const fullProps = {};

  const pureComponent = (props) => shallow(
    <ResendConfirmationForm {...props} />
  );

  it('matches component snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
