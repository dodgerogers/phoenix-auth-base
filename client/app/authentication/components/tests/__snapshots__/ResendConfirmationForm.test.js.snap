// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResendConfirmationForm matches component snapshot 1`] = `
<div
  className="resend-confirmation-form"
>
  <Form
    as="form"
    size="large"
  >
    <Field
      component={[Function]}
      fluid={true}
      icon="mail"
      iconPosition="left"
      name="email"
      placeholder="Email"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <Button
      as="button"
      color="teal"
      disabled={true}
      fluid={true}
      role="button"
      type="submit"
    >
      Submit
    </Button>
  </Form>
</div>
`;
